APE_Black_Formatting.rst

```diff
@@ -0,0 +1,286 @@
+mechanism allows for such cases to be explicitly excepted where the developer
+thinks the improved readability warrants it.
+
+History
+~~~~~~~
+This changes almost every Python file. Git blame can skip the commit, but git
+diff won't. But we are already doing other bulk changes that hit every file.
+
+Code correctness
+~~~~~~~~~~~~~~~~
+Black guarantees correctness of the code formatter to not change the code AST.
+
+Migration effort
+~~~~~~~~~~~~~~~~
+Enthusiastic champions of the effort will largely do the work, but we recognize
+and appreciate that additional effort will be required from maintainers that are
+neutral or against the adoption of Black.
+
+Backports / release branches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This is addressed in the Implementation section.
+
+New contributors
+~~~~~~~~~~~~~~~~
+New contributors may be more discouraged by more steps. However:
+
+This *replaces* the existing PEP8 check, so that it is not "another step" To
+make it better, leave very clear explicit messages in the status checks on GH
+(not putting up a barrier) Scikit-learn had a few issues with contributors but
+not at a level that caused regret about the decision to adopt black formatting.
+One developer described the change as a "net benefit".
+
+Removing the humanity
+~~~~~~~~~~~~~~~~~~~~~
+Some developers may feel that individual style and humanity gets lost in
+autoformatting. We hope to convey that humanity and individuality is expressed
+more fully in the elegance and clarity of the code logic and implementation. In
+most cases the placement of white space and line breaks can be viewed as a
+mechanistic process that humans need not control.
+
+Exception Policy
+^^^^^^^^^^^^^^^^
+As described above, there are cases Black formatting makes the code
+substantially more difficult to read or understand.. These should be treated
+explicitly as exceptions, using the ``#fmt: off``/``#fmt: on`` escape hatch for
+multi-line exceptions, or ``#fmt: skip`` for single-line exceptions. **This is
+allowed if the code author and PR reviewer(s) agree that the code line or block
+should be excluded from Black formatting.**
+
+
+Branches and pull requests
+--------------------------
+
+- `Add configuration for black autoformatter <https://github.com/astropy/astropy/pull/13253>`_
+- `Apply black to modeling <https://github.com/astropy/astropy/pull/13254>`_
+
+Implementation
+--------------
+
+All Python code in ``astropy`` is formatted with Black_, using its default
+settings, that is, 88 characters per line and double quotes. Note that, by
+design, most of Black_ is inflexible, and there is not a way to pick-and-choose
+some elements and not others of Black's format.
+
+Implementing this change requires:
+
+1. Updating the `coding style`_:
+
+   * Adding documentation about Black, with a particular emphasis on the rationale expressed here and the exception policy described above.
+   * Adding documentation about using pre-commit, ala `pandas pre-commit`_.
```

Comment:
Adding `pre-commit` documentation is already in progress as part of astropy/astropy#13552.

We already provide a `pre-commit` configuration and use it for our current "Code style checks" already, it just has not been properly documented in the `astropy` documentation.

Reply:
astropy/astropy#13552 has been merged, updating the documentation about how to use pre-commit is no longer necessary.